test-harness/modules/mocker/endpoints/open/bouncer.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) Mesosphere, Inc. See LICENSE file for details.
+
+"""Bouncer mock endpoint.
+"""
+
+import logging
+
+from exceptions import EndpointException
+from mocker.endpoints.recording import (
+    RecordingHTTPRequestHandler,
+    RecordingTcpIpEndpoint,
+)
+
+# pylint: disable=C0103
+log = logging.getLogger(__name__)
+
+
+# pylint: disable=R0903
+class BouncerHTTPRequestHandler(RecordingHTTPRequestHandler):
+    """A request hander class mimicking DC/OS IAM (Bouncer) operation.
+
+    Attributes:
+        USERS_PATH_PREFIX (str): API path under which users endpoint should
+            be available. The uid of the user is extracted from request path
+            itself.
+    """
+    USERS_PATH_PREFIX = '/acs/api/v1/users/'
+
+    def _calculate_response(self, base_path, *_):
+        """Answer the query for user data basing on the endpoint context.
+
+        Please refer to the description of the BaseHTTPRequestHandler class
+        for details on the arguments and return value of this method.
+
+        Raises:
+            EndpointException: request URL path is unsupported
+        """
+        ctx = self.server.context
+
+        with ctx.lock:
+            # copy.deepcopy() can also be used here, instead of locking
+            users = ctx.data['users']
+
+            if not base_path.startswith(self.USERS_PATH_PREFIX) or \
+                    base_path == self.USERS_PATH_PREFIX:
+                msg = "Path `{}` is not supported yet".format(self.USERS_PATH_PREFIX)
+                blob = msg.encode('utf-8', errors='backslashreplace')
```

Comment:
> I am not sure if base_path will always be utf8-encodable
It will, because `base_path` is a sequence of unicode code points, and UTF-8 is unicode-complete.